pkg/virt-handler/migration-proxy/migration-proxy.go

```diff
@@ -293,7 +303,13 @@ func (m *migrationProxyManager) StartSourceListener(key string, targetAddress st
 		filePath := SourceUnixFile(baseDir, proxyKey)
 
 		os.RemoveAll(filePath)
-		proxy := NewSourceProxy(filePath, targetFullAddr, m.serverTLSConfig, m.clientTLSConfig, key)
+		serverTLSConfig := m.serverTLSConfig
+		clientTLSConfig := m.clientTLSConfig
+		if m.config.GetMigrationConfiguration().DisableTLS != nil && *m.config.GetMigrationConfiguration().DisableTLS {
```

Comment:
Is there a chance that this value will change here during the creation of the proxies?

Reply:
:+1: Fixed. It is now only fetched once per migration and the same for all ports. I think the main source of  possible temporary migration failures on config changes remains, and that is that we don't know if source and target handlers are already seeing the same config. However I don't think that this is something we have to care about. A migration may not start but the next time it will.

Reply:
We could synchronize this via the migrationStatus and fail early with an understandable error message?

Reply:
True, but do you think it is relevant enough? I don't expect people to change this value often.

Reply:
> True, but do you think it is relevant enough? I don't expect people to change this value often.

yeah, we could have the migration controller put it in the vmi.Status.MigrationState field when it hands the migration off to the target/source handlers to be processed. That woudl have the added  benefit that we can see within the VMI status whether tls is used or not for current migration

as for whether we need this or not, as long as the migration fails if source/target have mixed usage of TLS being enabled or not, then that doesn't seem so bad. There are other things outside of our control that can cause migrations to fail as well today. I do like that by recording tls usage in the status that we wouldn't ever have to think about this edge case where mixed use of tls might exist.

all that said, I'll go with whatever you want to go with though @rmohr

Reply:
@vladikr if you don't insist, I would prefer to not add this to the status. What do you think? This should be a very seldom admin operation.

Reply:
I agree, it shouldn't be something that holds this PR from being merged. Perhaps, we could optimize this after this is merged.